docs/userguide/core-workflow.md

```diff
@@ -1,297 +1,73 @@
-# The Role of a Workflow
+# What is a Workflow?
 
-`Workflow` is a protocol (in Swift) and interface (in Kotlin) that defines the contract for a single
-node in the workflow hierarchy.
+A [Workflow](../../glossary#workflow-instance) defines the possible states and behaviors of components of a particular type.
+The overall state of a Workflow has two parts:
 
-=== "Swift"
-    ```Swift
-    public protocol Workflow: AnyWorkflowConvertible {
+* [Props](../../glossary#props), configuration information provided by whatever is running the Workflow
+* And the private [State](../../glossary#state) managed by the Workflow itself
 
-        associatedtype State
+At any time, a Workflow can be asked to transform its current Props and State into a [Rendering](../../glossary#rendering) that is suitable for external consumption.
+A Rendering is typically a simple struct with display data, and event handler functions that can enqueue [Workflow Actions](../../glossary#action) — functions that update State, and which may at the same time emit [Output](../../glossary#output) events.
 
-        associatedtype Output = Never
+![Workflow schematic showing State as a box with Props entering from the top, Output exiting from the top, and Rendering exiting from the left side, with events returning to the workflow via Rendering](../images/workflow_schematic.svg)
 
-        associatedtype Rendering
+A workflow Rendering usually serves as a view model in iOS or Android apps, although that is not a requirement.
+(In fact, this page includes no details about how platform specific UI code is driven.
+See [Workflow UI](../ui-concepts) for that discussion.
+And readers with an Android background should note the lower case _v_ and _m_ of "view model" — this notion has nothing to do with JetPack `ViewModel`.)
 
-        func makeInitialState() -> State
+![Workflow schematic with State type GameState, Props type Players, Rendering type GameScreen, and Output type GameOver. The workflow is receiving an onClick() event.](../images/game_workflow_schematic.svg)
```

Comment:
I'm not sure what to make of this image's placement.  It seems like it's meant to be a real-world example of the abstraction in the previous diagram, but since they're split by a paragraph, the connection is a little hard to make.

Reply:
Actually with the diagrams and supporting text in the next section, I'm not sure this image is necessary at all?

Reply:
WDYT with the new paragraph above? Does it work now, or should I delete it and the image?

Reply:
I like it.  It's nice baby steps.